### ISSUE TITLE
Ability to separate paramName from what gets attached to context/request object

### ISSUE THREAD

**Issue by ksmithut** [wilmoore/request-id.js#3]
It would be nice to have the ability to select the key that the request id gets attached to separate from the `req.query` parameter name.

The use case is that I would like to attach the id to `req.id` but keep the query parameter as `requestId` or disable the query parameter altogether.

I would be willing to submit a PR if that type of functionality goes along with your vision of this module.

**Comment by wilmoore** [comment]
Hey Keith,

Yeah, this is something I'd been thinking about supporting. I am actually going to be refactoring this soon such that it isn't so coupled to any specific frameworks but rather utilizes the intrinsic NodeJS http.Server API.

Your request is one that we'd get for free once that is done.

**Comment by ksmithut** [comment]
Sweet :) that'll be awesome. I'll close this and wait for that feature set.

**Comment by wilmoore** [comment]
nah, let's keep this open so we can close it out when the feature has landed :smile:
